src/clone.c

```diff
@@ -297,27 +390,53 @@ SYS_FUNC(clone3)
 	if (prefix_fun != tprint_value_changed_struct_begin)
 		tprint_struct_end();
 
+	if (show_namespace && tcp->u_rval >= 0) {
+		tcp->auxstr = get_namespace_auxstr(tcp->u_rval, arg.flags, tcp);
+		if (tcp->auxstr)
+			r_extra = RVAL_STR;
+	}
+
 out:
 	tprint_arg_next();
 	PRINT_VAL_U(size);
 
-	return RVAL_DECODED | RVAL_TID;
+	return RVAL_DECODED | RVAL_TID | r_extra;
 }
 
 
 SYS_FUNC(setns)
 {
-	printfd(tcp, tcp->u_arg[0]);
-	tprint_arg_next();
-	printflags(setns_types, tcp->u_arg[1], "CLONE_NEW???");
+	if (entering(tcp)) {
+		printfd(tcp, tcp->u_arg[0]);
+		tprint_arg_next();
+		printflags(setns_types, tcp->u_arg[1], "CLONE_NEW???");
+		return show_namespace ? 0: RVAL_DECODED;
+	}
+
+	int r_extra = 0;
+	if (show_namespace && tcp->u_rval == 0) {
+		tcp->auxstr = get_namespace_auxstr(tcp->pid, tcp->u_arg[1], NULL);
+		if (tcp->auxstr)
+			r_extra = RVAL_STR;
+	}
+	return RVAL_DECODED | r_extra;
 
-	return RVAL_DECODED;
 }
 
 SYS_FUNC(unshare)
 {
-	printflags64(unshare_flags, tcp->u_arg[0], "CLONE_???");
-	return RVAL_DECODED;
+	if (entering(tcp)) {
+		printflags64(unshare_flags, tcp->u_arg[0], "CLONE_???");
+		return show_namespace ? 0: RVAL_DECODED;
+	}
+
+	int r_extra = 0;
+	if (show_namespace && tcp->u_rval == 0) {
```

Comment:
The `SYS_FUNC(clone)` comment about `tcp->u_rval` also applies here.

Reply:
I think the most appropriate condition here would be `show_namespace && !syserror(tcp)`.

Reply:
fixed locally.